### PR TITLE
[2.4] Changes for detecting admin & cluster ownership

### PR DESCRIPTION
**Pull request by ryansann**
A role template with cluster ownership permissions can be either:
1. Builtin `Cluster Owner`
2. Have rules for "own" on "clusters" resources
3. Inherit from a role template that has 1. or 2.

When a user is added to a cluster with any of 1, 2, or 3, there should not be any namespaced roles created, this PR fixes an issue where that was happening and leading to errors. 

When a user is created and associated to a role, if they are associated to a global role with admin permissions that isn't admin, they are not getting the proper permissions / capabilities. Changes to globalrole_handler.go fix this issue. 

Previous related PR: 
- rancher/rancher#27978

### CODE REVIEW

**Comment by ryansann** [comment]
One thing I am not positive of here is whether the logic for detecting a Global Role with admin permissions is correct. Currently the `admin` role's rules looks like this: 
```yaml
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
- nonResourceURLs:
  - '*'
  verbs:
  - '*'
```
The changes I made in pkg/controllers/user/rbac/globalrole_handler.go look for the first rule in order to detect if roles other than `admin` should be treated as admin. Perhaps I should be checking for the second rule as well? Input appreciated

File: pkg/controllers/user/rbac/globalrole_handler.go [hunk 1]
```diff
@@ -70,9 +84,31 @@ func (c *grbHandler) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object
...
+	if err != nil {
+		return false, err
+	}
+
+	// global role is admin
+	if gr.Name == "admin" {
+		return true, nil
+	}
+
+	for _, rule := range gr.Rules {
```
[the remark this answers is not in the document]
Comment (ryansann) [reply pkg/controllers/user/rbac/globalrole_handler.go:104]: I changed this to check for both rules

File: pkg/controllers/user/rbac/globalrole_handler.go [hunk 2]
```diff
@@ -70,9 +84,42 @@ func (c *grbHandler) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object
...
 
+// isAdminRole detects whether a GlobalRole has admin permissions or not.
+func (c *grbHandler) isAdminRole(rtName string) (bool, error) {
+	gr, err := c.grLister.Get("", rtName)
+	if err != nil {
+		return false, err
+	}
+
+	// global role is admin
+	if gr.Name == "admin" {
```
Comment (mrajashree) [inline pkg/controllers/user/rbac/globalrole_handler.go:99]: can we also add a check for "Builtin" like we do for cluster-owner?

Comment (ryansann) [reply pkg/controllers/user/rbac/globalrole_handler.go:99]: Ahh yes, good point, let me add that

**Pull request merged**